### PR TITLE
coqPackages.extructures: 0.3.1 -> 0.4.0

### PR DESCRIPTION
**Pull request by arthuraa**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arthuraa** [comment]
@Zimmi48 @vbgl

File: pkgs/development/coq-modules/extructures/default.nix [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
   defaultVersion = with lib.versions; lib.switch [coq.coq-version ssreflect.version] [
+    { cases = [(range "8.17" "8.18") (isGe "2.0.0")  ]; out = "0.4.0"; }
     { cases = [(range "8.11" "8.18") (isGe "1.12.0") ]; out = "0.3.1"; }
     { cases = [(range "8.11" "8.14") (isLe "1.12.0") ]; out = "0.3.0"; }
     { cases = [(range "8.10" "8.12") (isLe "1.12.0") ]; out = "0.2.2"; }
   ] null;
 
   releaseRev = v: "v${v}";
 
+  release."0.4.0".sha256 = "sha256:1jbvawwmmczf9algdq81iqnrlmwwl8cxhwyrsi1kq3fmpvanx2f1";
```
Comment (vbgl) [inline pkgs/development/coq-modules/extructures/default.nix:20]:
```suggestion
  release."0.4.0".sha256 = "sha256-hItFO2XY2LTPSofPTKt3AfOEfiLliaYdzUXgDv4ea9Y=";
```

Comment (arthuraa) [reply pkgs/development/coq-modules/extructures/default.nix:20]: This is in base 64, right? For future reference, what is the easiest way to output the hash in this format?

Comment (vbgl) [reply pkgs/development/coq-modules/extructures/default.nix:20]: This is not a matter of format: the value is different.

Comment (arthuraa) [reply pkgs/development/coq-modules/extructures/default.nix]: @vbgl I am not sure I understand the issue. When I run

```
$ nix-prefetch-url --type sha256 https://github.com/arthuraa/extructures/archive/refs/tags/v0.4.0.tar.gz
```

I get the following output

```
path is '/nix/store/ka3l2rvylq06sy2fps1g25yb7w3lfgc7-v0.4.0.tar.gz'
1jbvawwmmczf9algdq81iqnrlmwwl8cxhwyrsi1kq3fmpvanx2f1
```

I took this value to fill in the `sha256` field in the format, but you seem to suggest that I should have computed the hash using some other command. How did you do it?

**Pull request merged**